src/parser.c

```diff
@@ -225,25 +234,33 @@ int ecs_parse_component_expr(
                 ecs_system_expr_oper_kind_t prev_oper_kind = oper_kind;
                 bptr = parse_complex_elem(bptr, &elem_kind, &oper_kind, &source);
                 if (!bptr) {
-                    ecs_abort(ECS_INVALID_EXPRESSION, sig);
+                    char err[50] = "/0";
+                    build_error_string(err, sig, ptr - sig, system_id);
+                    ecs_abort(ECS_INVALID_EXPRESSION, err);
                 }
 
                 if (oper_kind == EcsOperNot && prev_oper_kind == EcsOperOr) {
-                    ecs_abort(ECS_INVALID_EXPRESSION, sig);
+                    char err[50] = "/0";
+                    build_error_string(err, sig, ptr - sig, system_id);
```

Comment:
Suggested description:

"cannot use NOT (!) operator in an OR (|) expression"

Reply:
done